SlevomatCodingStandard/Sniffs/Namespaces/ReferenceUsedNamesOnlySniff.php

```diff
@@ -125,10 +129,33 @@ public function process(\PHP_CodeSniffer_File $phpcsFile, $openTagPointer)
 		$tokens = $phpcsFile->getTokens();
 
 		$referencedNames = ReferencedNameHelper::getAllReferencedNames($phpcsFile, $openTagPointer);
+
+		if ($this->allowFQNForCollidingClasses) {
+			$referencesIndex = array_combine(
+				array_map(
+					function (ReferencedName $referencedName) {
+						return $referencedName->getNameAsReferencedInFile();
+					},
+					$referencedNames
+				),
+				$referencedNames
+			);
+			$classPointer = $phpcsFile->findNext(TokenHelper::$typeKeywordTokenCodes, 0);
```

Comment:
I think you have to find all class pointers in the file in case there are more than one.